src/stan/model/indexing/rvalue.hpp

```diff
@@ -705,24 +721,14 @@ inline auto rvalue(StdVec&& v, const char* name, index_uni idx1,
  * @param[in] idx single index
  * @return Result of indexing array.
  */
-template <typename StdVec, require_std_vector_t<StdVec>* = nullptr>
+template <typename StdVec, require_std_vector_t<StdVec>* = nullptr,
+          require_not_t<std::is_reference<StdVec>>* = nullptr>
```

Comment:
What's the is reference for here?

Reply:
It looks like this doesn't allow for `StdVec&&`?

Reply:
By requiring `StdVec` to not be reference this prevents `StdVec&&` from being a lvalue reference. I think I can change this to make it a bit easier to read.